### PR TITLE
Enable unix build

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
@seanyen 
This is ready to be merged. Please have a look at the diff as I changed the recipe output stuff.

/cc @wolfv @traversaro

File: recipe/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+mkdir build
+cd build
+
+cmake .. \
+      -G "Ninja" \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_PREFIX_PATH=$PREFIX -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=True \
+      -DCMAKE_CXX_STANDARD=17
```
Comment (traversaro) [inline recipe/build.sh:12]: There is any specific reason why this was necessary? In theory if the project has a minimum required C++ version, this should be accounted as part of the CMake script.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -4,40 +4,38 @@
...
+    - {{ compiler('c') }}    # [not win]
+    - vs2017_win-64          # [win64]
+    - cmake
+    - ninja
+    - pkg-config
+  host:
+    - libignition-cmake2
+    - dlfcn-win32  # [win]
+  run:
+    - dlfcn-win32  # [win]
```
Comment (traversaro) [inline recipe/meta.yaml]: Probably unrelated, but as dlfcn-win32 has the run_exports (https://github.com/conda-forge/dlfcn-win32-feedstock/blob/master/recipe/meta.yaml#L15) probably we can drop this?

File: recipe/build.sh (hunk 1 above)
Comment (Tobias-Fischer) [reply recipe/build.sh:12]: I copied this blindly from https://github.com/conda-forge/libignition-common-feedstock/blob/master/recipe/build.sh

Maybe @wolfv can comment on this

Comment (Tobias-Fischer) [reply recipe/build.sh:12]: (or @seanyen actually as he has written this build script)

Comment (traversaro) [reply recipe/build.sh:12]: Ack, no problem then. However I think that everything should probably work even by removing that option.

Comment (wolfv) [reply recipe/build.sh:12]: Theoretically all of conda-forge uses c++17 by default. I would also guess that this isn't strictly necessary.

**Comment by wolfv** [comment]
Looks good to me.

**Comment by Tobias-Fischer** [comment]
Cool. Should we merge this one then? :)

**Review by traversaro — APPROVED** [review]

**Pull request merged**